lib/src/clojure_lsp/feature/completion.clj

```diff
@@ -422,20 +422,30 @@
                     :priority :clojurescript-core}
                    resolve-support))
 
-(defn ^:private with-java-items [matches-fn]
-  (concat
-    (->> common-sym/java-lang-syms
-         (filter (comp matches-fn str))
-         (map (fn [sym] {:label (str sym)
-                         :kind :class
-                         :detail (str "java.lang." sym)
-                         :priority :java-built-in})))
-    (->> common-sym/java-util-syms
-         (filter (comp matches-fn str))
-         (map (fn [sym] {:label (str sym)
-                         :kind :class
-                         :detail (str "java.util." sym)
-                         :priority :java-built-in})))))
+(defn ^:private with-java-definition-items [matches-fn cursor-value db]
+  ;; For performance reasons, we have thousands of class definitions usually
+  ;; we only consider it if user typed anything and is auto completing
+  (when (seq (str cursor-value))
+    (flatten
+      (into []
+            (comp
+              q/xf-analysis->java-class-definitions
+              (keep (fn [{:keys [class]}]
+                      (let [class-name* (delay (last (string/split class #"\.")))]
+                        (cond-> []
+
+                          (matches-fn class)
+                          (conj {:label (str class)
+                                 :kind :class
+                                 :priority :java-class-definitions})
+
+                          (and (string/starts-with? class "java.lang")
+                               (matches-fn @class-name*))
```

Comment:
As I read this, this lets you type `Str` and get a completion for `java.lang.String`... Is that correct? I think this would be more valuable if it filtered for "java" not "java.lang". The [java.lang package](https://docs.oracle.com/javase/7/docs/api/java/lang/package-summary.html) defines only 100 classes or so. They're used a lot, but a lot of the action is in other packges—java.util, java.concurrent, java.net, etc.

Reply:
No, the idea here is able to complete java lang classes by its name si cê Clojure allow you use it without importing, like `String`.
So we need to return these items without the package in the name, does that makes sense ?

Reply:
Ah, I see, yes, that's a nice feature. 👍